src/appleseed/renderer/meta/tests/test_sss.cpp

```diff
@@ -501,4 +600,178 @@ TEST_SUITE(Renderer_Modeling_BSSRDF_SSS)
             1.0e-16,        // ymin
             1.0e+1);        // ymax
     }
+
+    void plot_dirpole_and_nd_r(
+        const char*     filename,
+        const char*     title,
+        const double    rd,
+        const double    dmfp)
+    {
+        GnuplotFile plotfile;
+        plotfile.set_title(title);
+        plotfile.set_xlabel("r");
+        plotfile.set_ylabel("r R(r)");
+        plotfile.set_xrange(0.0, 4.0);
+        plotfile.set_yrange(0.001, 50.0);
+        plotfile.set_logscale_y();
+
+        auto_release_ptr<BSSRDF> dp_bssrdf(
+            DirectionalDipoleBSSRDFFactory().create("dirpole", ParamArray()));
+
+        DirectionalDipoleBSSRDFInputValues dp_values;
+        init_dirpole_bssrdf_values_rd_dmfp(rd, dmfp, 1.0, 0.0, dp_values);
+
+        auto_release_ptr<BSSRDF> nd_bssrdf(
+            NormalizedDiffusionBSSRDFFactory().create("norm_diff", ParamArray()));
+
+        NormalizedDiffusionBSSRDFInputValues nd_values;
+        nd_values.m_weight = 1.0;
+        nd_values.m_reflectance.set(rd);
+        nd_values.m_reflectance_multiplier = 1.0;
+        nd_values.m_dmfp = dmfp;
+        nd_values.m_dmfp_multiplier = 1.0;
+        nd_values.m_inside_ior = 1.0;
+        nd_values.m_outside_ior = 1.0;
+
+        const Vector3d normal(0.0, 1.0, 0.0);
+
+        ShadingPoint outgoing;
+        ShadingPointBuilder outgoing_builder(outgoing);
+        outgoing_builder.set_primitive_type(ShadingPoint::PrimitiveTriangle);
+        outgoing_builder.set_point(Vector3d(0.0, 0.0, 0.0));
+        outgoing_builder.set_shading_basis(Basis3d(normal));
+
+        ShadingPoint incoming;
+        ShadingPointBuilder incoming_builder(incoming);
+        incoming_builder.set_primitive_type(ShadingPoint::PrimitiveTriangle);
+        incoming_builder.set_shading_basis(Basis3d(normal));
+
+        const size_t N = 1000;
+        vector<Vector2d> nd_points, dp_points;
+
+        for (size_t j = 0; j < N; ++j)
+        {
+            const double r = max(fit<size_t, double>(j, 0, N - 1, 0.0, 4.0), 0.0001);
+            incoming_builder.set_point(Vector3d(r, 0.0, 0.0));
+
+            Spectrum result;
+
+            dp_bssrdf->evaluate(
+                &dp_values,
+                outgoing,
+                normal,
+                incoming,
+                normal,
+                result);
+            dp_points.push_back(Vector2d(r, result[0] /  r));
```

Comment:
I don't think this makes sense, the directional dipole's `evaluate()` method must (and does) return r * R(r), not just R(r).